### PR TITLE
Delete file(s) from remote server after successful upload to our Azure storage

### PR DESCRIPTION
**Pull request by somesylvie**
# Delete file(s) from remote server after successful upload to our Azure storage
Delete file(s) from remote server after successful upload to our Azure storage

Also updated the CRON for running the Azure function locally (which shouldn't impact real customers) and excluded `.txt` files from the `end-of-file-fixer` pre-commit hook because it was breaking the values of usernames and passwords locally

## Issue
https://github.com/CDCgov/trusted-intermediary/issues/1078

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added GoDocs where required
- [x] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

File: azure_functions/local.settings.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "AzureWebJobs.HttpExample.Disabled": "true",
     "AZURE_STORAGE_CONNECTION_STRING": "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:12000/devstoreaccount1;QueueEndpoint=http://127.0.0.1:12001/devstoreaccount1;",
     "POLLING_TRIGGER_QUEUE_NAME": "polling-trigger-queue",
-    "CA_DPH_POLLING_CRON": "0 30 9 * Jan Mon"
+    "CA_DPH_POLLING_CRON": "0 */1 * * * *"
```
Comment (pluckyswan) [inline azure_functions/local.settings.json:9]: Should we add a comment explaining why this is different from the other environments or is it too rudimentary?

File: src/zip/zip.go [hunk 2]
```diff
@@ -18,6 +18,12 @@ type ZipHandler struct {
 	zipClient        ZipClient
 }
 
+type ZipHandlerInterface interface {
```
Comment (pluckyswan) [inline src/zip/zip.go:21]: Is ZipClient a better name?

Comment (somesylvie) [reply src/zip/zip.go:21]: tragically that name is already taken (at the bottom of this file). I'm open to suggestions on renaming either/both

File: azure_functions/local.settings.json (hunk 1 above)
Comment (somesylvie) [reply azure_functions/local.settings.json:9]: json isn't allowed to have comments, but this CRON will run every minute, on the minute, and only in the `local` environment

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -11,6 +11,7 @@ repos:
       - id: detect-private-key
         exclude: (mock_credentials|_test\.go)
       - id: end-of-file-fixer
+        exclude: (\.txt)
```
Comment (somesylvie) [inline .pre-commit-config.yaml:14]: This pre-commit hook was adding extra whitespace to the mock_credentials files that contain fake usernames and passwords and was breaking them

File: src/orchestration/polling_message_handler.go [hunk 4]
```diff
@@ -17,7 +17,7 @@ func (receiver PollingMessageHandler) HandleMessageContents(message azqueue.Dequ
 	sftpHandler, err := sftp.NewSftpHandler()
 	if err != nil {
 		slog.Error("failed to create sftp handler", slog.Any(utils.ErrorKey, err))
-		// Don't return - just because polling is broken for one partner doesn't mean we should take down imports too
```
Comment (somesylvie) [inline src/orchestration/polling_message_handler.go:20]: I think when we originally wrote this comment, the `NewSftpHandler` call was happening somewhere else. It actually doesn't make sense here to not-return, hence the change

File: src/zip/zip.go (hunk 2 above)
Comment (jcrichlake) [reply src/zip/zip.go:21]: I like putting the word Interface or perhaps an I at the end of the name to mark it as an interface

**Review by pluckyswan — APPROVED** [review]

**Review by saquino0827 — APPROVED** [review]

File: src/sftp/sftp.go [hunk 5]
```diff
@@ -236,9 +250,12 @@ func (receiver *SftpHandler) copySingleFile(fileInfo os.FileInfo, index int, dir
 	err = receiver.blobHandler.UploadFile(fileBytes, blobPath)
 	if err != nil {
 		slog.Error("Failed to upload file", slog.Any(utils.ErrorKey, err))
+		return
 	}
 
 	slog.Info("About to consider whether this is a zip", slog.String("file name", fileInfo.Name()))
+
+	// TODO - if non-CA customers want us to retrieve non-zip files, will need to update this `if`
 	if strings.Contains(fileInfo.Name(), ".zip") {
```
Comment (jherrflexion) [inline src/sftp/sftp.go:259]: Consider flattening this on the next PR

File: src/sftp/sftp.go [hunk 6]
```diff
@@ -149,13 +159,17 @@ func getPublicKeysForSshClient(credentialGetter secrets.CredentialGetter) (ssh.S
 
 func (receiver *SftpHandler) Close() {
 	slog.Info("About to close SFTP handler")
-	err := receiver.sftpClient.Close()
-	if err != nil {
-		slog.Error("Failed to close SFTP client", slog.Any(utils.ErrorKey, err))
+	if receiver.sftpClient != nil {
+		err := receiver.sftpClient.Close()
```
[the remark this answers is not in the document]
Comment (somesylvie) [reply src/sftp/sftp.go:163]: nothing to flatten here

**Pull request merged**